src/components/Editor.vue

```diff
@@ -326,13 +326,17 @@ export default {
 		this.$syncService = null
 		this.$attachmentResolver = null
 	},
-	beforeDestroy() {
+	async beforeDestroy() {
 		if (!this.richWorkspace) {
 			window.removeEventListener('beforeprint', this.preparePrinting)
 			window.removeEventListener('afterprint', this.preparePrinting)
 		}
 		unsubscribe('text:image-node:add', this.onAddImageNode)
 		unsubscribe('text:image-node:delete', this.onDeleteImageNode)
+		if (this.dirty) {
+			const timeout = new Promise((resolve) => setTimeout(resolve, 2000))
+			await Promise.any([timeout, this.$syncService.save()])
```

Comment:
Maybe I'm missing something but wouldn't this be called after the text session has been closed already?

Reply:
No. The text session is closed afterwards where we destroy all the providers - one of which is the `SyncServiceProvider` which will close the session on destruction.
Only if the save takes longer than the timeout the provider will be destroyed never the less.

Reply:
Right, makes sense. I just got confused by the close/_close methods of the sync service and how they would be triggered as I assumed that the viewer close calls them, but that doesn't seem to be the case. 👍